<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DevWindow</class>
 <widget class="QMainWindow" name="DevWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>682</width>
    <height>403</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>DsmCP - Dev</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="groupBoxBoardStatus">
      <property name="title">
       <string>Status</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0" colspan="2">
        <widget class="QError" name="frameBoardError"/>
       </item>
       <item row="1" column="0">
        <widget class="QRegisterFlagDev" name="frameBoardStatus1">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QRegisterFlagDev" name="frameBoardStatus2">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="tabMotor">
       <attribute name="title">
        <string>Motor</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="1">
         <widget class="QGroupBox" name="groupBoxTorque">
          <property name="title">
           <string>Torque</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayoutTorque">
           <item>
            <widget class="QTorque" name="widgetTorque">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QActionButton" name="pushButtonStop">
          <property name="text">
           <string>PushButton</string>
          </property>
          <property name="icon">
           <iconset resource="res/img.qrc">
            <normaloff>:/img/32/stop.png</normaloff>:/img/32/stop.png</iconset>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QActionButton" name="pushButtonStart">
          <property name="text">
           <string>PushButton</string>
          </property>
          <property name="icon">
           <iconset resource="res/img.qrc">
            <normaloff>:/img/32/start.png</normaloff>:/img/32/start.png</iconset>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBoxSpeed">
          <property name="title">
           <string>Speed</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayoutSpeed">
           <item>
            <widget class="QSpeed" name="widgetSpeed">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabBoard">
       <attribute name="title">
        <string>Board</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBoxSystemMode">
          <property name="title">
           <string>System mode</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QSystemModeDev" name="frameSystemMode">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QGroupBox" name="groupBoxAnalogInputs">
          <property name="title">
           <string>Analog inputs</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout"/>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabDebug">
       <attribute name="title">
        <string>Debug</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0" rowspan="3">
         <widget class="QGroupBox" name="groupBoxDebugValues">
          <property name="title">
           <string>Values</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4"/>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QGroupBox" name="groupBoxDebugDAC">
          <property name="title">
           <string>DAC Signals</string>
          </property>
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="labelDebugDac0">
             <property name="text">
              <string>DAC #0</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="comboBoxDebugDac0"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="labelDebugDac1">
             <property name="text">
              <string>DAC #1</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="comboBoxDebugDac1"/>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QFrame" name="frame">
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QActionButton" name="pushButtonRefreshDebug">
             <property name="text">
              <string>PushButton</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="1">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>682</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuSystem">
    <property name="title">
     <string>System</string>
    </property>
    <addaction name="actionLink"/>
    <addaction name="actionConnect"/>
    <addaction name="actionRefresh"/>
    <addaction name="actionAutoRefresh"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuMotor">
    <property name="title">
     <string>Motor</string>
    </property>
    <addaction name="actionMotorMode"/>
    <addaction name="actionMotorStart"/>
    <addaction name="actionMotorStop"/>
    <addaction name="actionLightEnabled"/>
    <addaction name="actionLightBlue"/>
    <addaction name="actionMotorReverse"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionHelp"/>
    <addaction name="actionAbout"/>
    <addaction name="actionRegistersView"/>
   </widget>
   <widget class="QMenu" name="menuWorkspace">
    <property name="title">
     <string>Workspace</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuBoard">
    <property name="title">
     <string>Board</string>
    </property>
    <addaction name="actionFirmReset"/>
    <addaction name="actionAPISendFirmware"/>
    <addaction name="actionAboutBoard"/>
    <addaction name="actionRefreshDebug"/>
    <addaction name="actionRefreshAnalogInputs"/>
   </widget>
   <addaction name="menuSystem"/>
   <addaction name="menuBoard"/>
   <addaction name="menuWorkspace"/>
   <addaction name="menuMotor"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBarSystem">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionLink"/>
   <addaction name="actionConnect"/>
   <addaction name="actionRefresh"/>
  </widget>
  <widget class="QToolBar" name="toolBarMotor">
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionMotorStart"/>
   <addaction name="actionMotorStop"/>
   <addaction name="actionMotorReverse"/>
   <addaction name="actionLightEnabled"/>
   <addaction name="actionLightBlue"/>
  </widget>
  <action name="actionLink">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res/img.qrc">
     <normaloff>:/img/32/unlink.png</normaloff>
     <normalon>:/img/32/link.png</normalon>
     <activeoff>:/img/32/link.png</activeoff>
     <activeon>:/img/32/unlink.png</activeon>:/img/32/unlink.png</iconset>
   </property>
   <property name="text">
    <string>Link</string>
   </property>
   <property name="toolTip">
    <string>Link DCP to board</string>
   </property>
  </action>
  <action name="actionMotorStart">
   <property name="icon">
    <iconset>
     <normalon>:/img/32/start.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Start</string>
   </property>
   <property name="toolTip">
    <string>Start the motor</string>
   </property>
   <property name="shortcut">
    <string>F2</string>
   </property>
  </action>
  <action name="actionLightEnabled">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res/img.qrc">
     <normaloff>:/img/32/light-off.png</normaloff>
     <normalon>:/img/32/light-on.png</normalon>
     <activeoff>:/img/32/light-on.png</activeoff>
     <activeon>:/img/32/light-off.png</activeon>:/img/32/light-off.png</iconset>
   </property>
   <property name="text">
    <string>Light Enabled</string>
   </property>
   <property name="toolTip">
    <string>Enable/disable Light</string>
   </property>
   <property name="shortcut">
    <string>F4</string>
   </property>
  </action>
  <action name="actionLightBlue">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res/img.qrc">
     <normaloff>:/img/32/light.png</normaloff>
     <normalon>:/img/32/light-blue.png</normalon>
     <activeoff>:/img/32/light-blue.png</activeoff>
     <activeon>:/img/32/light.png</activeon>:/img/32/light.png</iconset>
   </property>
   <property name="text">
    <string>BlueLight</string>
   </property>
   <property name="toolTip">
    <string>Switch white / blue light</string>
   </property>
   <property name="shortcut">
    <string>F9</string>
   </property>
  </action>
  <action name="actionMotorReverse">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="res/img.qrc">
     <normaloff>:/img/32/clockwise.png</normaloff>
     <normalon>:/img/32/counterclockwise.png</normalon>
     <activeoff>:/img/32/counterclockwise.png</activeoff>
     <activeon>:/img/32/clockwise.png</activeon>:/img/32/clockwise.png</iconset>
   </property>
   <property name="text">
    <string>Reverse</string>
   </property>
   <property name="toolTip">
    <string>Reverses the motor rotation direction</string>
   </property>
   <property name="shortcut">
    <string>F7</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F1</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAutoRefresh">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normalon>:/img/32/autorefresh.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>AutoRefresh</string>
   </property>
  </action>
  <action name="actionUnlink">
   <property name="icon">
    <iconset>
     <normalon>:/img/32/unlink.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Unlink</string>
   </property>
   <property name="toolTip">
    <string>Unlink DCP from board</string>
   </property>
  </action>
  <action name="actionRefresh">
   <property name="icon">
    <iconset>
     <normalon>:/img/32/refresh.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Refresh</string>
   </property>
   <property name="toolTip">
    <string>Refresh</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F5</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="icon">
    <iconset>
     <normalon>:/img/32/help.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Help</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionMotorStop">
   <property name="icon">
    <iconset>
     <normalon>:/img/32/stop.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="toolTip">
    <string>Stop the motor</string>
   </property>
   <property name="shortcut">
    <string>F3</string>
   </property>
  </action>
  <action name="actionFirmReset">
   <property name="text">
    <string>Firmware Reset</string>
   </property>
   <property name="toolTip">
    <string>Reset the firmware settings</string>
   </property>
  </action>
  <action name="actionAPISendFirmware">
   <property name="text">
    <string>Send Firmware</string>
   </property>
  </action>
  <action name="actionAboutBoard">
   <property name="text">
    <string>About board...</string>
   </property>
  </action>
  <action name="actionMotorMode">
   <property name="text">
    <string>Motor mode...</string>
   </property>
   <property name="toolTip">
    <string>Show motor mode dialog</string>
   </property>
   <property name="shortcut">
    <string>F10</string>
   </property>
  </action>
  <action name="actionRegistersView">
   <property name="text">
    <string>Registers view</string>
   </property>
   <property name="shortcut">
    <string>Shift+F1</string>
   </property>
  </action>
  <action name="actionConnect">
   <property name="icon">
    <iconset resource="res/img.qrc">
     <normaloff>:/img/32/key.png</normaloff>:/img/32/key.png</iconset>
   </property>
   <property name="text">
    <string>Connect...</string>
   </property>
   <property name="toolTip">
    <string>Connect</string>
   </property>
  </action>
  <action name="actionRefreshDebug">
   <property name="icon">
    <iconset resource="res/img.qrc">
     <normaloff>:/img/32/refresh.png</normaloff>:/img/32/refresh.png</iconset>
   </property>
   <property name="text">
    <string>Refresh debug</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionRefreshAnalogInputs">
   <property name="text">
    <string>Refresh Analog Inputs</string>
   </property>
   <property name="toolTip">
    <string>Refresh Analog Inputs</string>
   </property>
   <property name="shortcut">
    <string>Shift+F5</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QActionButton</class>
   <extends>QPushButton</extends>
   <header>..widget.qactionbutton</header>
  </customwidget>
  <customwidget>
   <class>QSpeed</class>
   <extends>QFrame</extends>
   <header>..widget.qspeed</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QError</class>
   <extends>QFrame</extends>
   <header>..widget.qerror</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QTorque</class>
   <extends>QFrame</extends>
   <header>..widget.qtorque</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QSystemModeDev</class>
   <extends>QFrame</extends>
   <header>..widget.qsystemmode</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QRegisterFlagDev</class>
   <extends>QFrame</extends>
   <header>..widget.qregisterflag</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="res/img.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>DevWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
